### DISCUSSION TITLE
Error when using taskkill

### DISCUSSION THREAD

**Discussion by midpl0x** [LizardByte/community#135] [Sunshine]
Im unable to run Lego Star Wars: The Complete Saga using command. I have made it launch with detached. 

Now, when I exit the app on my client, I wish for the application to close. Through research, I've found: 

taskkill /im legostarwarssaga.exe /f /fi "STATUS eq RUNNING" 

Should do the trick. This, I have also confirmed manually through cmd. But when I add the command to undo, the app fails to launch. 

What am I doing wrong?

Thanks

**Comment by ReenigneArcher** [comment]
Do you have a `do` command?

**Reply by midpl0x** [reply]
Edit: I misunderstood the question. 

But no, there aren't any do command. I wouldn't even know what to put there

**Reply by ReenigneArcher** [reply]
try this:

```batch
echo "sunshine" > NUL
```

**Reply by midpl0x** [reply]
Just tried it, still no cigar. 

![billede](https://user-images.githubusercontent.com/50043134/211737490-09e0d1c7-52c6-4eff-85ce-06e337d2ba7c.png)

This is how I have it set up. Everything should be fine, right?

Edit: I appreciate you taking your time to help me.

**Reply by eTaurus** [reply]
Hi!
I'm faring well with
```
cmd echo whatever
```
as my `do` command. As `undo` I use
```
taskkill /IM "yourgame.exe" /F
```
Since I use Launchbox/BigBox to start my games I add a pair of `do` and `undo` commands for every game.
![Screenshot 2023-01-13 220001](https://user-images.githubusercontent.com/24603992/212418212-715d397c-5eb4-48e2-87fe-19f9749005ff.png)

**Reply by midpl0x** [reply]
Oh, hi eTaurus. I'll give it a go! I hadn't seen your reply. 

Btw. Do you happen to know of a way to launch a game using the "steam steam://rungameid/xxxxx" in windows? 

The issue is, that if I try to launch Deep Rock Galactic with the .exe, I won't be properly connected, and can't play the game

**Reply by eTaurus** [reply]
This is how I do it:
![Screenshot 2023-01-21 155929](https://user-images.githubusercontent.com/24603992/213872780-82a8db20-d5d3-4476-b098-d4386d5343e7.png)

**Reply by midpl0x** [reply]
I must be an idiot then. It looks (almost) like the way I tried it. 

I'll give it a go, thanks!

**Reply by eTaurus** [reply]
Where do you get the Game ID?

**Reply by midpl0x** [reply]
I open the steam store in my browser, navigate to the correct game, and it's shown in the address bar

**Comment by Silas888** [comment]
I found that you have to type in the entire path of the game in the working directory in Sunlight. For example if you have the following path in the command field: "D:\Games\Lego Star Wars\ LEGOStarWarsSaga.exe" you would enter the following as the working directory: "D:\Games\Lego Star Wars" 
After you do this, you'll note that all of the Lego games will work properly.

**Reply by midpl0x** [reply]
Hi Silas, 

I just tried it your suggestion. It did not work, unfortunately. Even with working directory, I still cannot launch the game with command. It seems detached is the only way to go. However, I am still unable to input any do or undo commands. Doing this, results in a connection error.

**Reply by Silas888** [reply]
When you say "input commands" I assume you mean a gamepad? If this is the case, have you installed ViGEmBus? https://github.com/ViGEm/ViGEmBus/releases/tag/v1.21.442.0

**Reply by midpl0x** [reply]
Oh no, Sorry. Controller works just fine. I meant the "do" and "undo" commands. 

Using the regular "command" to launch the game, results in connection error. So I have to use detached. But when launching the game with detached, the game doesn't close, when I exit the app in Moonlight.

**Reply by Silas888** [reply]
Yes, I feel you. When you launch a game through a detached command, you lose the ability to close it in Moonlight. I know that launching Steam games in particular require a detached command. I certainly hope that this will be addressed in the future sunshine releases I know that this was not a requirement within GeForce Experience.

**Reply by hoonlight** [reply]
There doesn't seem to be a way yet.

I think the only way for now is to create a script that closes the game and add it as another application to Sunshine.

[a comment by an automated account is not shown]
